### PR TITLE
adicionando o tile carousel

### PR DESCRIPTION
**Pull request by claytonc**

### CODE REVIEW

File: src/brasil/gov/tiles/tiles/templates/carousel.pt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                                     data-content-type obj/portal_type;
+                                     data-content-uuid python:view.get_uuid(obj)">
+                  <tal:image define="thumbnail python:view.thumbnail(obj);"
+                             condition="thumbnail">
+                    <a tal:attributes="href python:view.get_url(obj)">
+                      <img tal:attributes="src thumbnail/url|string:${item/absolute_url}/download;
+                                           width thumbnail/width;
+                                           height thumbnail/height;
+                                           class view/get_image_position;
+                                           alt python:view.get_alt(obj);" />
```
Comment (hvelarde) [inline src/brasil/gov/tiles/tiles/templates/carousel.pt]: in the future we'll need to review this attribute as the alternate text has to be entered explicitly.

File: src/brasil/gov/tiles/tiles/carousel.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    index = ViewPageTemplateFile('templates/carousel.pt')
+    short_name = _(u'msg_short_name_carousel', default=u'Carousel')
+
+    is_configurable = False
+    is_droppable = True
+    is_editable = True
+    limit = 4 * 3
+
+    @property
+    def tile_description(self):
```
[the remark this answers is not in the document]
Comment (claytonc) [reply src/brasil/gov/tiles/tiles/carousel.py]: ok

File: src/brasil/gov/tiles/tiles/carousel.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        :rtype: unicode
+        """
+        # First we get the title for the item itself
+        title = item.Title()
+        uuid = self.get_uuid(item)
+        data_mgr = ITileDataManager(self)
+        data = data_mgr.get()
+        uuids = data['uuids']
+        if uuid in uuids:
+            if uuids[uuid].get('custom_title', u''):
```
[the remark this answers is not in the document]
Comment (claytonc) [reply src/brasil/gov/tiles/tiles/carousel.py]: copy: 
https://github.com/collective/collective.cover/blob/master/src/collective/cover/tiles/carousel.py#L69

File: src/brasil/gov/tiles/tiles/carousel.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        :rtype: unicode
+        """
+        # First we get the url for the item itself
+        description = item.Description()
+        uuid = self.get_uuid(item)
+        data_mgr = ITileDataManager(self)
+        data = data_mgr.get()
+        uuids = data['uuids']
+        if uuid in uuids:
+            if uuids[uuid].get('custom_description', u''):
```
[the remark this answers is not in the document]
Comment (claytonc) [reply src/brasil/gov/tiles/tiles/carousel.py]: copy:
https://github.com/collective/collective.cover/blob/master/src/collective/cover/tiles/carousel.py#L89

File: src/brasil/gov/tiles/tiles/carousel.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        # First we get the url for the item itself
+        url = getattr(item, 'remoteUrl', item.absolute_url())
+        if item.portal_type in get_types_use_view_action_in_listings():
+            url += '/view'
+        uuid = self.get_uuid(item)
+        data_mgr = ITileDataManager(self)
+        data = data_mgr.get()
+        uuids = data['uuids']
+        if uuid in uuids:
+            if uuids[uuid].get('custom_url', u''):
```
[the remark this answers is not in the document]
Comment (claytonc) [reply src/brasil/gov/tiles/tiles/carousel.py]: copy:
https://github.com/collective/collective.cover/blob/master/src/collective/cover/tiles/carousel.py#L110

File: src/brasil/gov/tiles/tiles/templates/carousel.pt [hunk 6]
```diff
@@ -0,0 +1,57 @@
+<html xmlns="http://www.w3.org/1999/xhtml"
+      xmlns:metal="http://xml.zope.org/namespaces/metal"
+      xmlns:tal="http://xml.zope.org/namespaces/tal"
+      xmlns:i18n="http://xml.zope.org/namespaces/i18n"
+      i18n:domain="brasil.gov.tiles">
+  <body tal:define="is_empty view/is_empty">
+    <p tal:condition="python: is_empty and view.is_compose_mode()" i18n:translate="">
+      Please add up to <span i18n:name="limit_objects" tal:replace="view/limit"/> objects to the tile.
```
[the remark this answers is not in the document]
Comment (claytonc) [reply src/brasil/gov/tiles/tiles/templates/carousel.pt]: ok

File: src/brasil/gov/tiles/tiles/templates/carousel.pt (hunk 1 above)
Comment (claytonc) [reply src/brasil/gov/tiles/tiles/templates/carousel.pt]: ok

File: src/brasil/gov/tiles/tiles/carousel.py [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+    @property
+    def tile_description(self):
+        return self.data['tile_description']
+
+    @property
+    def switch_text(self):
+        return self.data['switch_text']
+
+    def results(self):
```
[the remark this answers is not in the document]
Comment (claytonc) [reply src/brasil/gov/tiles/tiles/carousel.py]: ok

File: src/brasil/gov/tiles/tiles/carousel.py [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        data_mgr = ITileDataManager(self)
+        data = data_mgr.get()
+        uuids = data['uuids']
+        if uuid in uuids:
+            if uuids[uuid].get('custom_url', u''):
+                # If we had a custom url set, then get that
+                url = uuids[uuid].get('custom_url')
+        return url
+
+    def is_empty(self):
```
[the remark this answers is not in the document]
Comment (claytonc) [reply src/brasil/gov/tiles/tiles/carousel.py]: ok

File: src/brasil/gov/tiles/tiles/carousel.py [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+        try:
+            next(self.results())
+        except StopIteration:
+            return True
+        return None
```
[the remark this answers is not in the document]
Comment (claytonc) [reply src/brasil/gov/tiles/tiles/carousel.py]: ok

File: src/brasil/gov/tiles/tiles/carousel.py [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+
+
+class ICarouselTile(IListTile):
+    """Display a carousel of items."""
+
+    tile_description = schema.Text(title=_(u'Tile Description'), required=False)
+    form.omitted('tile_description')
+    form.no_omit(ITileEditForm, 'tile_description')
+
+    switch_text = schema.TextLine(title=_(u'Switch Text'), required=False)
```
[the remark this answers is not in the document]
Comment (hvelarde) [reply src/brasil/gov/tiles/tiles/carousel.py]: this seems to be a "gambiarra" to join two carousels into one using tabs; we need to discuss if this feature is really needed in IDG.

**Comment by claytonc** [comment]
Fiz um novo branch para este tile

**Pull request closed without merging**